api/commands/run.ts

```diff
@@ -110,15 +112,24 @@ export class Runner {
     this.runResult.timing = timer.end();
 
     if (this.graph.runConfig && this.graph.runConfig.useRunCache) {
+      await this.adapter.prepareStateMetadataTable();
+
+      await this.adapter.deleteStateMetadata(this.graph.actions);
+
       const successfulActions = this.runResult.actions
-        .filter(action => action.status === dataform.ActionResult.ExecutionStatus.SUCCESSFUL)
+        .filter(action =>
+          [
+            dataform.ActionResult.ExecutionStatus.SUCCESSFUL,
+            dataform.ActionResult.ExecutionStatus.CACHE_SKIPPED
+          ].includes(action.status)
+        )
         .map(action =>
           this.graph.actions.find(executionAction => action.name === executionAction.name)
         );
 
       // Currently, we don't support caching for operations (and any dependents)
       await this.adapter.persistStateMetadata(
-        successfulActions.filter(action => action.type !== "operation")
+        successfulActions.filter(action => action && action.type !== "operation")
```

Comment:
how can `action` possibly be falsey? AFAICS `action` comes from a lookup in the `ExecutionGraph` for the action with the same name in `RunResult`. i can't see how an action would appear in `RunResult` if it weren't in `ExecutionGraph` - indeed, if this happens, we should definitely blow up as this implies something has gone Really Wrong

Reply:
seems not needed.